packages/scaffold-config/test/unit/detect.spec.ts

```diff
@@ -1,100 +1,199 @@
 import { expect } from 'chai'
+import fs from 'fs-extra'
 import { e2eProjectDirs } from '@packages/frontend-shared/cypress/e2e/support/e2eProjectDirs'
 import { detect } from '../../src'
 import Fixtures from '@tooling/system-tests/lib/fixtures'
 import path from 'path'
-import fs from 'fs-extra'
 
 export async function scaffoldMigrationProject (project: typeof e2eProjectDirs[number]) {
   Fixtures.removeProject(project)
 
   await Fixtures.scaffoldProject(project)
 
-  return fs.readJSONSync(path.join(Fixtures.projectPath(project), 'package.json'))
+  return path.join(Fixtures.projectPath(project))
+}
+
+interface DepToFake {
+  dependency: string
+  version: string
+}
+
+interface DevDepToFake {
+  devDependency: string
+  version: string
+}
+
+/**
+ * The way we detect dependencies is by using resolve-from (https://www.npmjs.com/package/resolve-from).
+ * In these unit tests, we don't want to actually run `npm install`, since it is slow,
+ * so this function fakes that the dependencies are installed by creating pretend dependency like this:
+ * `node_modules/<dependency>/package.json.
+ * Inside `package.json` we add the minimal:
+ *
+ * {
+ *   "version": "5.0.0",
+ *   "main": "index.js"
+ * }
+ *
+ * We have some real e2e tests that actually run `npm install`.
+ * Those are in launchpad/cypress/e2e/scaffold-component-testing.cy.ts.
+ */
+function fakeDepsInNodeModules (cwd: string, deps: Array<DepToFake | DevDepToFake>) {
```

Comment:
Do we even still need to fake the deps? Shouldn't the system tests infra take care of this for us?

Reply:
We've got real ones with real deps here: https://github.com/cypress-io/cypress/blob/lmiller1990/UNIFY-1402/packages/launchpad/cypress/e2e/scaffold-component-testing.cy.ts. They are kind of slow.

We also have more more here testing the correct install deps string is generated: https://github.com/cypress-io/cypress/blob/lmiller1990/UNIFY-1402/packages/data-context/test/unit/sources/WizardDataSource.spec.ts#L13

The reason I added these is so I could easily test many cases with less common configurations like 

```json
{
  "dependencies": {
    "vite": "version: '2.5.0-alpha.4"
  }
}
```

Without the overhead of actually running a yarn install, since this is supposed to be a unit test for the `detect` function. I see some value in more "unit-y" tests, so we can easily express less common cases without having to spin up an entire launchpad workflow test, eg:

```js
it(`React with Vite using pre-release version`, async () => {
    const projectPath = await scaffoldMigrationProject('react-vite-ts-unconfigured')

    fakeDepsInNodeModules(projectPath, [
      { devDependency: 'vite', version: '2.5.0-alpha.4' },
      { dependency: 'react', version: '17.0.0' },
    ])

    const actual = detect(projectPath)

    expect(actual.framework.type).to.eq('react')
    expect(actual.bundler.type).to.eq('vite')
  })
```

I could go either way, but to answer your original question:

- yes, we have e2e tests using the system tests infrastructure
- no, we don't strictly need to fake the deps, but I am not sure system-tests infra is the right tool for testing `detect`, which has quite a lot of conditionals that are more concisely expressed in the unit tests here.

I don't have a super strong opinion here, but these unit tests really helped with getting all the edge cases nailed, like `package.json` with less common configs, like `1.0.0-alpha.10`. What do you think?